main/contracts/sourcifyContract.ts

```diff
@@ -0,0 +1,83 @@
+import log from 'electron-log'
+import { Response } from 'node-fetch'
+import { JsonFragment } from '@ethersproject/abi'
+import { hexToNumberString } from 'web3-utils'
+import type { ContractSource } from '.'
+import { fetchWithTimeout } from '../../resources/utils/fetch'
+
+interface SourcifySourceCodeResponse {
+  status: string,
+  files: SourcifySourceCodeFile[]
+}
+
+interface SourcifySourceCodeFile {
+  name: string
+  path: string
+  content: string
+}
+
+interface SourcifyMetadataFileContent {
+  compiler: { version: string },
+  language: string,
+  output: {
+    abi: JsonFragment[],
+    devdoc: Partial<{
+      details: string
+      title: string
+    }>
+  },
+  settings: Partial<{
+    compilationTarget: {
+      [K: string]: string
+    },
+    evmVersion: string
+    metadata: { 
+      [K: string]: string
+    }
+  }>,
+  sources: {
+    [K: string]: {
+      [J: string]: string | string[]
+    }
+  },
+  version: number
+}
+
+function getEndpointUrl (contractAddress: Address, chainId: string) {
+  return `https://sourcify.dev/server/files/any/${hexToNumberString(chainId)}/${contractAddress}`
+}
+
+async function parseResponse <T>(response: Response): Promise<T | undefined> {
+  if (response?.status === 200 && (response?.headers.get('content-type') || '').toLowerCase().includes('json')) {
+    return await response.json()
+  }
+  return Promise.resolve(undefined)
+}
+
+async function fetchSourceCode (contractAddress: Address, chainId: string): Promise<SourcifyMetadataFileContent | undefined> {
+  const endpointUrl = getEndpointUrl(contractAddress, chainId)
+  
+  try {
+    const res = await fetchWithTimeout(endpointUrl, {}, 4000)
+    const parsedResponse = await parseResponse<SourcifySourceCodeResponse>(res as Response)
+
+    return parsedResponse && ['partial', 'full'].includes(parsedResponse.status) ? JSON.parse(parsedResponse.files[0].content) : Promise.reject(`Contract ${contractAddress} not found in Sourcify`)
+  } catch (e) {
+    log.warn((e as Error).name === 'AbortError' ? 'Sourcify request timed out' : 'Unable to parse Sourcify response', e);
```

Comment:
semicolon

Reply:
This release and the merge of the formatting PR can't come soon enough